xonsh/platform.py

```diff
@@ -96,12 +115,68 @@ def is_readline_available():
 #
 
 DEFAULT_ENCODING = sys.getdefaultencoding()
+""" Default string encoding. """
+
+
+def scandir(path):
+    """ This is either `os.scandir` on Python 3.5+ or a function providing a
+    compatibility layer for it.
+    This function is recommended for iterations over directory entries at a
+    significantly higher speed than `os.listdir` on Python 3.5+.
+    """
+    pass
+
+
+if PYTHON_VERSION_INFO < (3, 5, 0):
+    from pathlib import Path
+
+    class DirEntry:
```

Comment:
Did you copy this directly from somewhere? If so it needs a comment with the original copyright.